### PR TITLE
Tests/gps pi

### PR DESCRIPTION
**Pull request by yashasvid**
## Title
Added tests for the GPS module to improve coverage

## Description
The purpose of the PR is to improve the code coverage for the GPS module. 

## Types of Changes
_Put an `x` in the boxes that apply_

- [x] Feature (non-breaking change which adds functionality)
- [ ] Bug Fix (non-breaking change that fixes an issue)
- [ ] Breaking Change (feature/fix that causes existing features to not work as expected)
- [ ] Documentation

## Checklist

- [x] I have read the [contribute]contributing<link> doc
- [x] Classes, scripts, and environment variables follow existing naming convention
- [x] Lint and Unit tests pass locally
- [x] New features on hardware have been tested on a local Raspberry Pi
- [x] Mention new programs/binaries if any must be installed along with this change
- [x] Mention new environment variables if any have been added to hardware/env file
- [x] Test coverage should not drop more than 3%
- [x] Any new required python modules are added to the requirements.txt

### CODE REVIEW

File: hardware/tests/test_gps_pi.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            gps_reader = GPSReader("GPS_HAT_LOG_FILE")
+            mock_port.assert_called_with(
+                os.environ["GPS_PORT"], os.environ["GPS_BAUDRATE"],
+            )
+
+            self.assertTrue(gps_reader.logging is not None)
+            self.assertTrue(gps_reader.logging.name == "GPS_HAT_LOG_FILE")
+            self.assertIsInstance(gps_reader.logging, Logger)
+
+    @patch("hardware.gpsPi.gps_reader.date_str_with_current_timezone")
```
Comment (ab7289) [inline hardware/tests/test_gps_pi.py:61]: it doesn't look like the date_str_with_current_timezone function is actually used in gps_reader, is this for the `data["date"]` object at line 52?

Comment (yashasvid) [reply hardware/tests/test_gps_pi.py:61]: Yes, it is for the data["date"] object. Trying to understand what is wrong here.

Comment (ab7289) [reply hardware/tests/test_gps_pi.py:61]: i think you're data might be out of date with hardware-team, try running updating your remotes and then run either `git merge origin/hardware-team` or `git rebase origin/hardware-team`

Comment (yashasvid) [reply hardware/tests/test_gps_pi.py:61]: okay

Comment (ab7289) [reply hardware/tests/test_gps_pi.py:61]: always a good idea before you're going to PR to update your remotes and check to see if there is new code in your base branch (hardware-team here)

Comment (yashasvid) [reply hardware/tests/test_gps_pi.py:61]: Agreed.

**Comment by yashasvid** [comment]
I am planning to add more tests to improve the coverage. Let me know if you have any feedback on the existing tests.

File: hardware/gpsPi/gps_reader.py [hunk 2]
```diff
@@ -49,7 +48,7 @@ def get_geolocation(self):
                 "latitude": latitude_decimal,
                 "longitude": longitude_decimal,
             }
-            data["date"] = datetime.now(timezone.utc).astimezone().isoformat()
+            data["date"] = date_str_with_current_timezone()
```
Comment (ab7289) [inline hardware/gpsPi/gps_reader.py:51]: I think @MichaelLally made this change in his previous PR, are we sure that it should be changed back?

Comment (yashasvid) [reply hardware/gpsPi/gps_reader.py:51]: Yes, I asked him. He said we can.

Comment (ab7289) [reply hardware/gpsPi/gps_reader.py:51]: oh great!

**Pull request merged**